ApollosProject/apollos-apps
# feat: Add Comment Feature

## DESCRIPTION

### What does this PR do, or why is it needed?

Returns an Add Comment Feature when a Content Item has Comments enabled.

### What design trade-offs/decisions were made?

Prompts are returned in the feature, with default values defined in the content-item data-source.

### How do I test this PR?

With this query, anywhere you see a CommentListFeature, you should also see an AddCommentFeature:

```
query {
  contentChannels {
    childContentItemsConnection {
      edges {
        node {
          id
          title
          ... on WeekendContentItem {
            featureFeed {
              features {
                id
                ... on CommentListFeature {
                  id
                }
                ... on AddCommentFeature {
                  id
                  addPrompt
                  initialPrompt
                  relatedNode {
                    id
                    __typename
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

### What automated tests did you write?

I moved the assertions against the CommentListFeature into a new test along with the AddCommentFeature assertions.

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
